custom/ewsghana/reports/maps.py

```diff
@@ -47,7 +46,11 @@ def _get_icon_and_color(self, value):
         }[value]
 
     def get_data(self):
-        for loc in self.locations:
+        filtered_locations = [
+            location for location in self.locations
+            if SQLProduct.objects.get(product_id=self.active_product.get_id) in location.sql_location.products
```

Comment:
it looks like this db lookup will be called once per location - can you pull it out of the list comprehension?